python/pypto/runtime/bench.py

```
@@ -128,6 +130,45 @@ def by_name(self) -> dict[str, "TraceSpan"]:
+        span = self.by_name().get(_SPAN_DEVICE)
+        return span.dur_us if span is not None else 0.0
+
+    @property
+    def host_wall_us(self) -> float:
+        """Host ``run_prepared`` duration (µs), 0 if absent."""
+        span = self.by_name().get(_SPAN_HOST)
+        return span.dur_us if span is not None else 0.0
+
+    @property
+    def effective_us(self) -> float:
+        """L2 **Effective** window (µs): the orch∪sched merged on-device window.
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF002)